qiskit/providers/aer/library/save_matrix_product_state.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Simulator instruction to save matrix product state.
+"""
+
+from qiskit.circuit import QuantumCircuit
+from .save_data import SaveSingleData, default_qubits
+
+
+class SaveMatrixProductState(SaveSingleData):
+    """Save matrix product state instruction"""
+    def __init__(self, num_qubits,
+                 label="matrix_product_state",
+                 pershot=False,
+                 conditional=False):
+        """Create new instruction to save the matrix product state.
+
+        Args:
+            num_qubits (int): the number of qubits
+            label (str): the key for retrieving saved data from results.
+            pershot (bool): if True save the mps for each
+                            shot of the simulation [Default: False].
+            conditional (bool): if True save data conditional on the current
+                                classical register values [Default: False].
+
+        .. note::
+
+            This save instruction must always be performed on the full width of
+            qubits in a circuit, otherwise an exception will be raised during
+            simulation.
+        """
+        super().__init__('save_matrix_product_state',
+                         num_qubits,
+                         label,
+                         pershot=pershot,
+                         conditional=conditional)
+
+
+class SaveMatrixProductStateDict(SaveSingleData):
```

Comment:
I dont think we need this. The MPS is already a custom structure

Reply:
removed.